src/bridging/providers/across/util.ts

```diff
@@ -29,43 +37,117 @@ export function getTokenAddress(tokenSymbol: string, chainConfig: AcrossChainCon
+    amountsAndCosts: toAmountsAndCosts(request, slippageBps, suggestedFees),
+    quoteTimestamp: Number(suggestedFees.timestamp),
+    expectedFillTimeSeconds: Number(suggestedFees.expectedFillTimeSec),
+    suggestedFees,
+  }
+}
+
+function toAmountsAndCosts(
+  request: QuoteBridgeRequest,
+  slippageBps: number,
+  suggestedFees: SuggestedFeesResponse
+): BridgeQuoteAmountsAndCosts {
+  const { amount, sellTokenDecimals, buyTokenDecimals } = request
+
+  // Get the amounts before fees
+  const sellAmountBeforeFeeBig = getBigNumber(amount, sellTokenDecimals)
+  const sellAmountBeforeFee = sellAmountBeforeFeeBig.big
+  const buyAmountBeforeFee = getBigNumber(sellAmountBeforeFeeBig.num, buyTokenDecimals).big // Sell and buy token should be the same asset for current implementation, but technically they can have different decimals
+
+  // Apply the fee to the buy amount (sell amount doesn't change)
+  const totalRelayerFeePct = BigInt(suggestedFees.totalRelayFee.pct)
+  const buyAmountAfterFee = applyPctFee(buyAmountBeforeFee, totalRelayerFeePct)
+
+  // Calculate the fee
+  const feeSellToken = sellAmountBeforeFee - applyPctFee(sellAmountBeforeFee, totalRelayerFeePct)
+  const feeBuyToken = buyAmountBeforeFee - buyAmountAfterFee
+  // TODO: Do we need to use any of the other fees, or they are included in totalRelayFee? I know 'lpFee' fee is, as stated in the docs, but not sure about the others.
   // const relayerCapitalFee = suggestedFees.relayerCapitalFee
   // const relayerGasFee = suggestedFees.relayerGasFee
   // const lpFee = suggestedFees.lpFee
 
+  // Apply slippage
+  const buyAmountAfterSlippage = applyBps(buyAmountAfterFee, slippageBps)
+
   return {
-    suggestedFees,
-    buyAmount: applyFee(amount, suggestedFees.totalRelayFee.pct),
-    feeBps: pctToBps(suggestedFees.totalRelayFee.pct),
-    slippageBps: 0, // TODO: Can I set slippage to zero? are quotes enforced?
+    beforeFee: {
+      sellAmount: sellAmountBeforeFee,
+      buyAmount: buyAmountBeforeFee, // Assuming the price is 1:1 (before fee). This is because we are exchanges the same asset
+    },
+    afterFee: {
+      sellAmount: sellAmountBeforeFee, // Sell amount does't change (fee is applied to the buy amount)
+      buyAmount: buyAmountAfterFee,
+    },
+    afterSlippage: {
+      sellAmount: sellAmountBeforeFee, // Sell amount does't change (slippage is applied to the buy amount)
+      buyAmount: buyAmountAfterSlippage,
+    },
+
+    costs: {
+      bridgingFee: {
+        feeBps: pctToBps(totalRelayerFeePct),
+        amountInSellCurrency: feeSellToken,
+        amountInBuyCurrency: feeBuyToken,
+      },
+    },
+    slippageBps,
   }
 }
 
 /**
- * pct is a string that represents a percentage.
+ * Assert that a percentage is valid.
+ */
+function assertValidPct(pct: bigint): void {
+  if (pct > PCT_100_PERCENT || pct < 0n) {
+    throw new Error('Fee cannot exceed 100% or be negative')
+  }
+}
+
+/**
+ * pct is a represents a percentage.
```

Comment:
Does this sentence belong here?
Also, there's a typo:

```suggestion
 * pct represents a percentage.
```